app/src/main/java/ch/hikemate/app/model/route/Hike.kt

```diff
@@ -0,0 +1,43 @@
+package ch.hikemate.app.model.route
+
+import com.google.firebase.Timestamp
+
+/**
+ * Represents a hike route with associated information.
+ *
+ * Some data about the hike might not be available immediately, because they are costly to compute
+ * and should only be obtained if absolutely necessary. For example [waypoints] or [elevation]. See
+ * [DeferredData] for more information about how this works.
+ *
+ * @param id The unique ID of the hike.
+ * @param isSaved Whether the hike has been saved by the user.
+ * @param plannedDate The date at which the user plans to go on the hike. Null if the user did not
+ * save the date or saved it without a specific date.
+ * @param name Human-readable name of the hike.
+ * @param description Human-readable description of the hike.
+ * @param bounds Minimum and maximum latitude/longitude of the hike on a map.
+ * @param waypoints List of geographical points that form the line of the hike.
+ * @param elevation List of [Double] values that give the elevation (in meters) along the hike.
+ * @param distance Total length of the hike (in kilometers).
+ * @param estimatedTime How much time the hike is estimated to take in order to complete it. The
+ *   value is a [Double] that represents the number of hours it would take.
+ * @param elevationGain The cumulative elevation gain of the hike (in meters).
+ * @param difficulty Estimation of how difficult the hike is on a three-levels scale.
+ * @see [ch.hikemate.app.utils.RouteUtils]
+ * @see [HikeDifficulty]
+ * @see [DeferredData]
+ */
+data class Hike(
+    val id: String,
+    val isSaved: Boolean,
+    val plannedDate: Timestamp?,
+    val name: String?,
+    val description: String?,
```

Comment:
Question: If I understand correctly, This would never be waiting to retrieve.
In the case in which we would like to only save the id of the hike when saved. Then all of these values should also be `DeferredData`. What do you think ?

Reply:
True, I'll set `name` and `description` as `DeferredData`, however I will assume for now that `plannedDate` is still stored with `isSaved` in the database. It can be adapted to make "saved" and "planned" two completely different things, as discussed during yesterday's meeting, but this task is already gigantic and I probably don't need to account for this yet.

Reply:
Actually setting `name` to a `DeferredData` would be too complicated, since we actually store the name with the ID of the hike when saving it. If we stop storing the name with the saved hike, we'll have to change it at that point.